src/server/locals.js

```diff
@@ -0,0 +1,173 @@
+/* globals Map */
+var inspector = require('inspector');
+var async = require('async');
+
+function Locals(config) {
+  if (!(this instanceof Locals)) {
+    return new Locals(config);
+  }
+
+  this.config = config;
+
+  this.initSession();
+}
+
+Locals.prototype.initSession = function() {
+  if (Locals.session) { return; }
+
+  Locals.session = new inspector.Session();
+  Locals.session.connect();
+  Locals.currentErrors = new Map();
+
+  Locals.session.on('Debugger.paused', ({ params }) => {
+    if (params.reason == 'promiseRejection' || params.reason == 'exception') {
+      var key = params.data.description;
+      Locals.currentErrors.set(key, params);
+
+      // Set the max size of the current errors array.
+      // The value should be large enough to preserve each of the errors in
+      // the current cause chain.
+      var CURRENT_ERRORS_MAX_SIZE = 4;
+
+      if (Locals.currentErrors.size > CURRENT_ERRORS_MAX_SIZE) {
+        var firstKey = Locals.currentErrors.keys()[0];
+        Locals.currentErrors.delete(firstKey);
+      }
+    }
+  });
+
+  Locals.session.post('Debugger.enable', (_err, _result) => {
+    Locals.session.post('Debugger.setPauseOnExceptions', { state: 'all'}, (_err, _result) => {
+    });
+  });
+}
+
+Locals.prototype.currentLocalsMap = function() {
+  if (Locals.currentErrors.size) {
+    return new Map(Locals.currentErrors);
+  }
+}
```

Comment:
Updated to always return a map. I've kept the check at: https://github.com/rollbar/rollbar.js/blob/aaafe97927a0c0741300725330269f98dd84985c/src/server/parser.js#L313 

because `item.localsMap` will be undefined when locals is disabled.